stan/math/rev/mat/functor/integrate_1d_tsc.hpp

```diff
@@ -0,0 +1,179 @@
+#ifndef STAN_MATH_PRIM_ARR_FUNCTOR_integrate_1d_tsc_HPP
+#define STAN_MATH_PRIM_ARR_FUNCTOR_integrate_1d_tsc_HPP
+
+#include <stan/math/prim/scal/meta/is_constant_struct.hpp>
+#include <stan/math/prim/mat/fun/value_of.hpp>
+#include <stan/math/rev/scal/fun/value_of.hpp>
+#include <stan/math/rev/scal/fun/to_var.hpp>
+#include <stan/math/rev/mat/fun/to_var.hpp>
+#include <stan/math/rev/arr/fun/to_var.hpp>
+#include <stan/math/prim/scal/err/check_finite.hpp>
+
+#include <stan/math/prim/scal/meta/scalar_seq_view.hpp>
+#include <stan/math/prim/scal/meta/operands_and_partials.hpp>
+#include <stan/math/rev/mat/functor/de_integrator.hpp>
+#include <functional>
+#include <cmath>
+#include <ostream>
+#include <vector>
+
+namespace stan {
+
+  namespace math {
+
+    /**
+     * Return the numeric integral of a function f given its gradient g.
+     *
+     * @tparam T Type of f.
+     * @tparam G Type of g.
+     * @param f a functor with signature
+     * double (double, std::vector<T_param>) or with signature
+     * double (double, T_param) where the first argument is one being
+     * integrated and the second one is either an extra scalar or vector
+     * being passed to f.
+     * @param g a functor with signature
+     * double (double, std::vector<T_param>, int, std::ostream&) or with
+     * signature double (double, T_param, int, std::ostream&) where the
+     * first argument is onebeing integrated and the second one is
+     * either an extra scalar or vector being passed to f and the
+     * third one selects which component of the gradient vector
+     * is to be returned.
+     * @param a lower limit of integration, must be double type.
+     * @param b upper limit of integration, must be double type.
+     * @param tre target relative error.
+     * @param tae target absolute error.
+     * @param param aditional parameters to be passed to f.
+     * @param msgs stream.
+     * @return numeric integral of function f.
+     */
+    template <typename F, typename G, typename T_param>
+    inline typename scalar_type<T_param>::type
+    integrate_1d_tsc_tscg(const F& f,  const G& g,  const double a,
```

Comment:
There is no explanation of what `tsc` or `tscg` are.  The integration algorithm and conditions on it also need to be described.  What kinds of functions can be integrated?  Do the bounds need to be finite?  Are the absolute and relative error such that both must hold or is one enough?

Reply:
I think what cause the confusion was that this was meant to be `integrate_1d_tscg`, but a search and replace mistake made it `integrate_1d_tsc_tscg`. Fixed.